test/test_constrmatrix.py

```diff
@@ -0,0 +1,18 @@
+from .helper import check_evaluation, session
+import pytest
+
+
+@pytest.mark.parametrize(
+    ("str_expr", "str_expected", "fail_msg"),
+    [
+        (
+            "DiagonalMatrix[a + b]",
+            "DiagonalMatrix[a + b]",
+            "Leaves unchanged? Adapted from #> code that was in tensor.py",
```

Comment:
Maybe this could be `DiagonalMatrix is only defined for lists`.

Reply:
Ok. I looked at the doc and this is the way it is because clearly the expression `a + b` where neither a nor b has been defined is not an expression. It seems weird to me that someone would this property in this way.

Reply:
Well, WL is a symbolic language, thus it seems logical that almost any symbol can be attached to a value, so I guess there isn't another way to test this.

Reply:
A simpler test is use a single literal value like 1, or "a"